test/unittests/eof_example_test.cpp

```diff
@@ -0,0 +1,290 @@
+        //               |                                               |
+        //    version    |                                 Header terminator
+        //    |          |___________                      |             |_________
+        "EF00 01 01 0004 02 0001 0004 03 0001 0014 04 0000 00 00 80 0002 5F 5F EE00"
+        //       |‾‾‾‾‾‾              |‾‾‾‾‾‾‾‾‾‾‾ |‾‾‾‾‾‾    |‾‾‾‾‾‾‾‾‾
+        //       |                    |            Header: data section 0 bytes long
+        //       |                    |                       |
+        //       Header: types section 4 bytes long           Types section: first code section
+        //                            |                       0 inputs, non-returning,
+        //                            |                       max stack height 2
+        //       Header: 1 subcontainer 20 bytes long
+        //
+        //////////////////
+        // Deployed container (contract doing nothing, see Example 1)
+        "EF00 01 01 0004 02 0001 0001 04 0000 00 00 80 0000 00");
+
+    // Tests the code is valid EOF and when called with initcodes creates a new contract.
+    tx.to = To;
+    pre.insert(*tx.to, {
+                           .code = factory,
+                       });
+
+    // Address of the newly created contract is calculated using the salt, initcontainer hash and
+    // deployer address.
+    expect.post[0x5ea44d9b32a04ae2c15fe4fa8ebf9a8a5a1e7e89_address].exists = true;
+    expect.post[*tx.to].exists = true;
+}
+
+TEST_F(state_transition, eof_examples_data)
+{
+    // # Example 6
+    //
+    // A basic EOF contract with a data section being used to load a byte of data onto the stack.
+
+    rev = EVMC_PRAGUE;
+
+    // clang-format off
+    const auto eof_code = bytecode(
+        //                   Code section: DATALOADN onto the stack the first word of data, STOP
+        //                                                  |
+        //               Header: 1 code section 4 bytes long 
+        //               |                                  |
+        //    version    |                    Header terminator       Data section
+        //    |          |___________         |             |________ |_________________________________________________________________
+        "EF00 01 01 0004 02 0001 0004 04 0021 00 00 80 0001 D10000 00 454F462068617320736F6D65206772656174206578616D706C6573206865726521"
+        //       |‾‾‾‾‾‾              |‾‾‾‾‾‾    |‾‾‾‾‾‾‾‾‾
+        //       |                    Header: data section 33 bytes long
+        //       |                               |
+        //       Header: types section 4 bytes long
+        //                                       |
+        //                                       Types section: first code section 0 inputs,
+        //                                       non-returning, max stack height 1
+    );
+    // clang-format on
+
+    // Tests the code is valid EOF and does nothing.
```

Comment:
Maybe put the loaded value in storage?

Reply:
Hm, nah. The purpose here is to demonstrate the data section, let's let the example keep its primary focus.